afs-ext-base/src/main/java/com/powsybl/afs/ext/base/ImportedCaseBuilder.java

```diff
@@ -134,14 +146,37 @@ public ImportedCase build() {
         if (context.getStorage().getChildNode(context.getFolderInfo().getId(), name).isPresent()) {
             throw new AfsException("Parent folder already contains a '" + name + "' node");
         }
-
-        // create project file
-        NodeInfo info = context.getStorage().createNode(context.getFolderInfo().getId(), name, ImportedCase.PSEUDO_CLASS, "", ImportedCase.VERSION,
-                new NodeGenericMetadata().setString(ImportedCase.FORMAT, importer.getFormat()));
-
-        // store case data
-        importer.copy(dataSource, new AppStorageDataSource(context.getStorage(), info.getId(), info.getName()));
-
+        NodeInfo info = null;
+        if (dataSource != null) {
+
+            // create project file
+            info = context.getStorage().createNode(context.getFolderInfo().getId(), name, ImportedCase.PSEUDO_CLASS, "", ImportedCase.VERSION,
+                    new NodeGenericMetadata().setString(ImportedCase.FORMAT, importer.getFormat()));
+            // store case data
+            importer.copy(dataSource, new AppStorageDataSource(context.getStorage(), info.getId(), info.getName()));
+        } else {
+            importer = Importers.findImporter(dataStore, name, importersLoader, context.getProject().getFileSystem().getData().getShortTimeExecutionComputationManager(), importConfig);
```

Comment:
Related to the comment above about using a data pack instead of data store. Here the `name` is actually the future name of the `ImportedCase` in AFS, not the name of the network file in the data store, so it will probably not work.

With the data pack, it should be simpler, because you should be able to simply do something like
```java
dataPack.copyTo(new AppStorageDataStore(context.getStorage(), info.getId()));
```

Reply:
Ok, I should have read the part below before :)

But the comment is still right: with a data pack as input, you will not even need to check for an importer. Just copy the data and get the format ID from the pack.